frontend/public/components/factory/details.tsx

```diff
@@ -89,6 +89,7 @@ export type DetailsPageProps = {
   icon?: React.ComponentType<{ obj: K8sResourceKind }>;
   getResourceStatus?: (resource: K8sResourceKind) => string;
   children?: React.ReactNode;
+  customKind?: string;
```

Comment:
Why is this customKind needed?

Reply:
I am reading from a secret resource to display the details of a Helm Release. If I don't send this customKind prop, then it will display the Secret kind in the Resource Icon instead of Helm Release